components/sysutils/nut/Makefile

```diff
@@ -0,0 +1,178 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 Jim Klimov
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=	nut
+COMPONENT_VERSION=	2.7.4
+COMPONENT_SUMMARY=	Network UPS Tools (NUT)
+COMPONENT_DESCRIPTION=	Network UPS Tools (NUT) is a versatile power-device monitoring toolkit to facilitate safe shutdowns
+COMPONENT_SRC=	$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	\
+  sha256:d580915fdf7090655c1c7c98eb116b61952553f06f0039b1f93c9f5a13e2d36b
+COMPONENT_ARCHIVE_URL=	\
+  https://github.com/networkupstools/$(COMPONENT_NAME)/archive/v$(COMPONENT_VERSION).tar.gz
+COMPONENT_PROJECT_URL=	http://networkupstools.org
+COMPONENT_FMRI=	system/management/$(COMPONENT_NAME)
+COMPONENT_CLASSIFICATION=	System/Administration and Configuration
+COMPONENT_LICENSE=	GPLv2+
+COMPONENT_LICENSE_FILE=	COPYING
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/configure.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+CXXFLAGS.32=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+CFLAGS.32=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+
+CXXFLAGS+=$(CXXFLAGS.$(BITS))
+CFLAGS+=$(CFLAGS.$(BITS))
+
+LDFLAGS.32= -L$(USRLIBDIR) -R $(USRLIBDIR) $(LDFLAGS) -L$(GNULIB) -R $(GNULIB)
+LDFLAGS.64= -L$(USRLIBDIR64) -R $(USRLIBDIR64) $(LDFLAGS) -L$(GNULIB64) -R $(GNULIB64)
+
+### A lot of paths are defined and exported to pkg(5) tools below,
+### so we can flexibly set up location of certain NUT components.
+NUTUSER  =	nut
+PKG_OPTIONS +=	-D NUTUSER="$(NUTUSER)"
+NUTGROUP  =	nut
+PKG_OPTIONS +=	-D NUTGROUP="$(NUTGROUP)"
+
+# This is where config files are expected, INCLUDING (per convention from other
+# distros) example config files - e.g. not lost under /usr/shared/... locations
+# Also note we must not include the leading slash...
+NUTETCDIR  =	$(strip .../,,...$(ETCDIR)/nut)
+PKG_OPTIONS +=	-D NUTETCDIR="$(NUTETCDIR)"
+
+# Extra programs for NUT, not intended for direct user access (drivers, CGIs)
+NUTUSRDIR  =	$(strip .../,,...$(USRDIR)/nut)
+PKG_OPTIONS +=	-D NUTUSRDIR="$(NUTUSRDIR)"
```

Comment:
No strong opinion, and the variable is there to make fixes like this easy ;)

Some linux distros put them in `/lib/nut`, others leave them all in `(/usr)/bin` (IMHO the latter makes too much clutter with little benefit since the drivers are no longer directly usable without a common config snippet in `ups.conf`). In OI we also have some components with dedicated directories under `/usr` so I picked this as a pretty random choice.